File: Backend/game_of_codes_API/Services/Models/Clue.py
from Services.DatabaseServices.databaseCommands import databaseCommands
from Services.Models.GameSquare import GameSquare


class Clue:

    def __init__(self, gameID, playerName, guideName, clueText, relatedGameSquareList, color="B", numOfWordsHinted=0,
                 badness=-1, numOfWordsCorrectlyGuessed=0, clueID=0):

        self.gameID = gameID
        self.clueText = clueText
        self.guideName = guideName
        self.playerName = playerName
        self.relatedGameSquares = relatedGameSquareList
        self.badness = badness
        self.color = color
        self.numOfWordsHinted = numOfWordsHinted
        self.numOfWordsCorrectlyGuessed = numOfWordsCorrectlyGuessed

        if not clueID:
            self.clueID = databaseCommands.create_clue(self.gameID, self.guideName, self.playerName, self.clueText,
                                                   self.relatedGameSquares,self.numOfWordsHinted, self.color,
                                                    self.badness, self.numOfWordsCorrectlyGuessed)

        else:
            self.clueID = clueID



    def getRelatedGameSquares(self):
        return self.relatedGameSquares

    def updateClue(self, gameSquaresGuessed, badness, numCorrectlyGuessed):
        databaseCommands.update_clue(self.clueID, gameSquaresGuessed, badness, numCorrectlyGuessed)
        return True

    def serialiseClue(self):
        gameData = {
            'GameID': self.gameID,
            'clueText': self.clueText,
            'colour': self.color,
            'numOfHintedWords': self.numOfWordsHinted,
            'playerName': self.playerName,
            'guideName': self.guideName,
            'clueID': self.clueID,
            'badness': -1
        }
        return gameData

File: FrontEnd/game-of-codes-web/src/app/clue.ts
import { Word } from './word';

export class Clue {
    GameID: number;
    clueID: number;
    colour: string;
    wordsGuessed: Word[];
    badness: number;
    numOfHintedWords: number;
    numOfCorrectlyGuessed: number;
    clueText: string;
    playerName: string;
    guideName: string;
    teamTurn: string;

    constructor(gameId: number,clueTxt: string, colour: string, numOfWordsHinted:number, numOfCorrectlyGuessed:number, playerName: string, guideName: string){
        this.GameID = gameId;
        this.clueText = clueTxt;
        this.colour = colour;
        this.numOfHintedWords = numOfWordsHinted;
        this.numOfCorrectlyGuessed = numOfCorrectlyGuessed;
        this.playerName = playerName;
        this.guideName = guideName;
        this.wordsGuessed = [];
        this.badness = -1;
    }
}